File: mycollab-web/src/main/java/com/esofthead/mycollab/module/crm/view/activity/ActivityTaskUrlResolver.java
package com.esofthead.mycollab.module.crm.view.activity;

import com.esofthead.mycollab.common.UrlEncodeDecoder;
import com.esofthead.mycollab.eventmanager.EventBus;
import com.esofthead.mycollab.module.crm.domain.Meeting;
import com.esofthead.mycollab.module.crm.events.ActivityEvent;
import com.esofthead.mycollab.module.crm.view.CrmUrlResolver;

public class ActivityTaskUrlResolver extends CrmUrlResolver {
	public ActivityTaskUrlResolver() {
		this.addSubResolver("add", new TaskAddUrlResolver());
		this.addSubResolver("edit", new TaskEditUrlResolver());
		this.addSubResolver("preview", new TaskPreviewUrlResolver());
	}

	public static class TaskAddUrlResolver extends CrmUrlResolver {
		@Override
		protected void handlePage(String... params) {
			EventBus.getInstance().fireEvent(
					new ActivityEvent.TaskAdd(this, new Meeting()));
		}
	}

	public static class TaskEditUrlResolver extends CrmUrlResolver {
		@Override
		protected void handlePage(String... params) {
			String decodeUrl = UrlEncodeDecoder.decode(params[0]);
			int meetingId = Integer.parseInt(decodeUrl);
			EventBus.getInstance().fireEvent(
					new ActivityEvent.TaskEdit(this, meetingId));
		}
	}

	public static class TaskPreviewUrlResolver extends CrmUrlResolver {
		@Override
		protected void handlePage(String... params) {
			String decodeUrl = UrlEncodeDecoder.decode(params[0]);
			int accountId = Integer.parseInt(decodeUrl);
			EventBus.getInstance().fireEvent(
					new ActivityEvent.TaskRead(this, accountId));
		}
	}
}

File: mycollab-web/src/main/scala/com/esofthead/mycollab/module/crm/view/activity/ActivityTaskUrlResolver.scala
package com.esofthead.mycollab.module.crm.view.activity

import com.esofthead.mycollab.common.UrlTokenizer
import com.esofthead.mycollab.eventmanager.EventBusFactory
import com.esofthead.mycollab.module.crm.domain.Task
import com.esofthead.mycollab.module.crm.events.ActivityEvent
import com.esofthead.mycollab.module.crm.view.CrmUrlResolver

/**
 * @author MyCollab Ltd
 * @since 5.0.9
 */
class ActivityTaskUrlResolver extends CrmUrlResolver {
    this.addSubResolver("add", new TaskAddUrlResolver)
    this.addSubResolver("edit", new TaskEditUrlResolver)
    this.addSubResolver("preview", new TaskPreviewUrlResolver)

    class TaskAddUrlResolver extends CrmUrlResolver {
        protected override def handlePage(params: String*) {
            EventBusFactory.getInstance.post(new ActivityEvent.TaskAdd(this, new Task))
        }
    }

    class TaskEditUrlResolver extends CrmUrlResolver {
        protected override def handlePage(params: String*) {
            val meetingId: Integer = new UrlTokenizer(params(0)).getInt
            EventBusFactory.getInstance.post(new ActivityEvent.TaskEdit(this, meetingId))
        }
    }

    class TaskPreviewUrlResolver extends CrmUrlResolver {
        protected override def handlePage(params: String*) {
            val accountId: Integer = new UrlTokenizer(params(0)).getInt
            EventBusFactory.getInstance.post(new ActivityEvent.TaskRead(this, accountId))
        }
    }

}
